iwf/communication.py

```diff
@@ -66,3 +79,40 @@ def get_to_publishing_internal_channel(self) -> list[InterStateChannelPublishing
 
     def get_to_trigger_state_movements(self) -> list[StateMovement]:
         return self._state_movements
+
+    def get_internal_channel_size(self, channel_name):
+        registered_type = self._type_store.get(channel_name)
+
+        if registered_type is None:
+            for name, t in self._type_store.items():
+                if channel_name.startswith(name):
+                    registered_type = t
+
+        if registered_type is None:
+            raise WorkflowDefinitionError(
+                f"InternalChannel channel_name is not defined {channel_name}"
+            )
+
+        if (
+            self._internal_channel_infos is not None
+            and channel_name in self._internal_channel_infos
+        ):
+            server_channel_size = self._internal_channel_infos[channel_name].size
+        else:
+            server_channel_size = 0
+
+        if channel_name in self._to_publish_internal_channel:
+            buffer_channel_size = len(self._to_publish_internal_channel[channel_name])
+        else:
+            buffer_channel_size = 0
+
+        return server_channel_size + buffer_channel_size
+
+    def get_signal_channel_size(self, channel_name):
+        if (
```

Comment:
I think we also need to check the signal type

Reply:
Do you mean checking if the signal channel exists?

Communication object currently only stores internal channels in the `type_store`. I will need to extend it with `_signal_type_store` to do it

Reply:
Yes. Oh that’s right because communication never needed to access that